core/network/source_other.go

```diff
@@ -10,14 +10,60 @@ import (
 	"github.com/juju/errors"
 )
 
+// netAddr implements ConfigSourceAddr based on an address in string form.
+type netAddr struct {
+	addr  string
+	ip    net.IP
+	ipNet *net.IPNet
+}
+
+// newNetAddr returns a new netAddr reference
+// representing the input IP address string.
+func newNetAddr(a string) (*netAddr, error) {
+	res := &netAddr{
+		addr: a,
+	}
+
+	ip, ipNet, _ := net.ParseCIDR(a)
+	if ipNet != nil {
+		res.ipNet = ipNet
+	}
+
+	if ip == nil {
```

Comment:
if `err != nil` then we should attempt to parse the IP. We could then also use the error to improve the "unable to parse message"